src/extensions/timeseries/TimeWidgetNew/TimeWidgetNew.tsx

```diff
@@ -60,128 +59,128 @@ interface SetCurrentTime {
     currentTime: number | null
 }
 type TimeAction = ZoomTimeRangeAction | SetTimeRangeAction | TimeShiftFrac | SetCurrentTime | {type: 'gotoHome' | 'gotoEnd'}
-const timeReducer = (state: TimeState, action: TimeAction): TimeState => {
-
-    const fix = (s: TimeState): TimeState => {
-        if (s.numTimepoints === null) return s
-        let ret: TimeState = s
-        if (ret.currentTime !== null) {
-            if (ret.currentTime >= ret.numTimepoints) {
-                ret = {
-                    ...ret,
-                    currentTime: ret.numTimepoints - 1
-                }
-            }
-        }
-        if (ret.currentTime !== null) {
-            if (ret.currentTime < 0) {
-                ret = {
-                    ...ret,
-                    currentTime: 0
-                }
-            }
-        }
-        if (ret.timeRange !== null) {
-            if (ret.timeRange.max - ret.timeRange.min > ret.maxTimeSpan) {
-                ret = {
-                    ...ret,
-                    timeRange: {min: ret.timeRange.min, max: ret.timeRange.min + ret.maxTimeSpan}
-                }
-            }
-        }
-        if (ret.timeRange !== null) {
-            if (ret.timeRange.max > ret.numTimepoints) {
-                const dt = ret.numTimepoints - ret.timeRange.max
-                ret = {
-                    ...ret,
-                    timeRange: {min: ret.timeRange.min + dt, max: ret.timeRange.max + dt}
-                }
-            }
-        }
-        if (ret.timeRange !== null) {
-            if (ret.timeRange.min < 0) {
-                const dt = -ret.timeRange.min
-                ret = {
-                    ...ret,
-                    timeRange: {min: ret.timeRange.min + dt, max: ret.timeRange.max + dt}
-                }
-            }
-        }
-        return ret
-    }
+// const timeReducer = (state: TimeState, action: TimeAction): TimeState => {
+
+//     const fix = (s: TimeState): TimeState => {
+//         if (s.numTimepoints === null) return s
```

Comment:
This functionality was all moved to the `RecordingSelection`/`RecordingSelectionDispatch` right?

Reply:
yes